nats-concepts/jetstream/source_and_mirror.md

```diff
@@ -0,0 +1,69 @@
+# Source and Mirror
+
+When a stream is configured with a `source` or `mirror`, it will automatically and asynchronously replicate messages from the origin stream. There are several options when declaring the configuration.
+
+- `Name` - Name of the origin stream to source messages from.
+- `StartSeq` - An optional start sequence of the origin stream to start mirroring from.
+- `StartTime` - An optional message start time to start mirroring from. Any messages that are equal to or greater than the start time will be included.
+- `FilterSubject` - An optional filter subject which will include only messages that match the subject, typically including a wildcard. Note, this cannot be used with `SubjectTransforms`.
+- `SubjectTransforms` - An optional set of [subject transforms](../../running-a-nats-service/configuration/configuring_subject_mapping.md) to apply when sourcing messages from the origin stream. Note, in this context, the `Source` will act as a filter on the origin stream and the `Destination` can optionally be provided to apply a transform. Since multiple subject transforms can be used, disjoint subjects can be sourced from the origin stream while maintaining the order of the messages. Note, this cannot be used with `FilterSubject`.
+- `Domain` - An optional JetStream domain of where the origin stream exists. This is commonly used in a hub cluster and leafnode topology.
+
+The stream using a source or mirror configuration can have its own retention policy, replication, and storage type.
+
+{% hint style="info" %}
+* Changes to to the stream using source or mirror,e.g. deleting messages or publishing, do not reflect back on the origin stream from which the data was received.
+* Deletes in the origin stream are NOT replicated through a `source` or `mirror` agreement.
+{% endhint %}
+
+{% hint style="info" %}
+`Sources` is a generalization of the `Mirror` and allows for sourcing data from one or more streams concurrently. 
+If you require the target stream to act as a read-only replica:
+* Configure the stream without listen subjects **or**
+* Temporarily disable the listen subjects through client authorizations. 
+{% endhint %}
+
+## Source specific
+A stream defining `Sources` is a generalized replication mechanism and allows for sourcing data from **one or more streams** concurrently as well as allowing direct write/publish by clients. Essentially the source streams and client writes are aggregated into a single interleaved stream.
```

Comment:
You can even source the same stream more than once as long as the combination of the transform source + transform destination is unique for the stream being sourced more than once. 

(e.g. you can source foo twice, once with source "foo" and destination "bar" and once with source "foo" and destination "foobar" and indeed the message will be sourced twice and end up being 'duplicated' in the sourcing stream (though with two different subjects))